src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8753,14 +8753,21 @@ namespace WaterThermalTanks {
         Effic = WaterHeaterDesuperheater(DesuperheaterNum).HeatReclaimRecoveryEff;
 
         // store first iteration tank temperature and desuperheater mode of operation
-        if (FirstHVACIteration && !ShortenTimeStepSys) {
+        if (FirstHVACIteration && !ShortenTimeStepSys && WaterHeaterDesuperheater(DesuperheaterNum).FirstTimeThroughFlag) {
             // Save conditions from end of previous system timestep
             // Every iteration that does not advance time should reset to these values
             WaterThermalTank(WaterThermalTankNum).SavedTankTemp = WaterThermalTank(WaterThermalTankNum).TankTemp;
+            WaterThermalTank(WaterThermalTankNum).SavedSourceOutletTemp = WaterThermalTank(WaterThermalTankNum).SourceOutletTemp;
             WaterHeaterDesuperheater(DesuperheaterNum).SaveMode = WaterHeaterDesuperheater(DesuperheaterNum).Mode;
+            WaterHeaterDesuperheater(DesuperheaterNum).FirstTimeThroughFlag = false;
+        }
+
+        if (!FirstHVACIteration) {
+            WaterHeaterDesuperheater(DesuperheaterNum).FirstTimeThroughFlag = true;
```

Comment:
Should this be an `else` to the previous `if` statement?

Reply:
Good point, might be more clear if using ``else if`` ?

Reply:
So why didn't clang catch the formatting at lines 8763 - 8765> Should have been "} else if () {" on 1 line. @yzhou601 do you have clang enabled?

Reply:
Well, I just tried and clang did not change lines 8763-8765. So I guess this is OK.